src/hound/vcs/svn.go

```diff
@@ -0,0 +1,98 @@
+package vcs
+
+import (
+	"bytes"
+	"io"
+	"strings"
+	"io/ioutil"
+	"os/exec"
+	"path/filepath"
+	"encoding/xml"
+	"log"
+)
+
+
+type Info struct {
+	Entry Entry `xml:"entry"`
+}
+type Entry struct {
+	Revision string `xml:"revision,attr"`
+	Url string `xml:"url"`
+	RelativeUrl string `xml:"relative-url"`
+	Repository Repository `xml:"repository"`
+}
+type Repository struct {
+	Root string `xml:"root"`
+	Uuid string `xml:"uuid"`
+}
+
+func init() {
+	RegisterVCS("svn", &SvnDriver{})
+}
+
+type SvnDriver struct{}
+
+func pull(dir string) error {
+	cmd := exec.Command("svn", "update")
+	cmd.Dir = dir
+	return cmd.Run()
+}
+
+func (g *SvnDriver) HeadHash(dir string) (string, error) {
+	//TODO is there a better way to extract revision number in golang?
+	cmd := exec.Command(
+		"svn", "info", "--xml")
+	cmd.Dir = dir
+	r, err := cmd.StdoutPipe()
+	if err != nil {
+		return "", err
+	}
+	defer r.Close()
+
+	if err := cmd.Start(); err != nil {
+		return "", err
+	}
+
+	var buf bytes.Buffer
+
+	if _, err := io.Copy(&buf, r); err != nil {
+		return "", err
+	}
+
+	// TODO parse xml output
+	info := Info{}
+	if err := xml.Unmarshal(buf.Bytes(), &info); err != nil {
+		log.Printf("error: %v", err)
+		return "", err
+	}
+
+	hash := strings.TrimSpace(info.Entry.Repository.Uuid)
+	log.Printf("hash: %s", hash)
```

Comment:
Be sure to clean up the debugging stuff.

Also, there is a bit of a bad spot in our code right now in that we assume that hashes are 40 character hashes that consistent of [a-f0-9]. I need to clean it up, but for now it persists. See this:
https://github.com/etsy/Hound/blob/master/src/hound/searcher/searcher.go#L54